src/test/portmidienumeratortest.cpp

```diff
@@ -75,6 +75,19 @@ TEST_F(PortMidiEnumeratorTest, InputOutputPortsLinked) {
         "Pure Data Midi in 1",
         "Pure Data Midi out 1"));
 
+    ASSERT_FALSE(shouldLinkInputToOutput(
+            "Pure Data Midi in 1",
+            "Pure Data Midi out 2"));
```

Comment:
This seems to be paradoxical, or am I missing something?

Reply:
Can you elaborate? Devices with different numbers shouldn't be matched?

Reply:
Ah, yeah, thats what I was missing, I didn't realize the test above and below differed by their device number.